src/app/api/v1/admin/users/[primarySha1]/route.ts

```diff
@@ -117,15 +123,34 @@ export async function PUT(
       for (const action of actions) {
         switch (action) {
           case "subscribe": {
+            const sub = await getSubscriberByFxaUid(subscriber.fxa_uid!);
+            const currentFxAProfile = sub.fxa_profile_json as FxaProfile;
+            const subscriptions = new Set(currentFxAProfile.subscriptions);
+
+            subscriptions.add(MONITOR_PREMIUM_CAPABILITY);
+            // update fxa profile data to match subscription status
+            currentFxAProfile.subscriptions = Array.from(subscriptions);
+            await updateFxAProfileData(sub, JSON.stringify(currentFxAProfile));
+
             // activate and opt out profiles
-            await activateAndOptoutProfile(onerepProfileId);
+            await activateProfile(onerepProfileId);
+            await optoutProfile(onerepProfileId);
             logger.info("force_user_subscribe", {
               onerepProfileId,
               primarySha1,
             });
             break;
           }
           case "unsubscribe": {
+            const sub = await getSubscriberByFxaUid(subscriber.fxa_uid!);
+            const currentFxAProfile = sub.fxa_profile_json as FxaProfile;
+            const subscriptions = new Set(currentFxAProfile.subscriptions);
+
+            subscriptions.delete(MONITOR_PREMIUM_CAPABILITY);
+            // update fxa profile data to match subscription status
+            currentFxAProfile.subscriptions = Array.from(subscriptions);
```

Comment:
Heh, could also be done without converting to a set, mutating that set, and then converting it back to an array, see below.

But I have a similar concern here about it getting out of sync with the FxA data at the source of truth.

```suggestion
            // update fxa profile data to match subscription status
            currentFxAProfile.subscriptions = currentFxAProfile.subscriptions.filter(sub => sub !== MONITOR_PREMIUM_CAPABILITY);
```